packages/calcite-components/src/components/card-group/card-group.tsx

```diff
@@ -0,0 +1,251 @@
+import {
+  connectInteractive,
+  disconnectInteractive,
+  InteractiveComponent,
+  InteractiveContainer,
+  updateHostInteraction,
+} from "../../utils/interactive";
+import { createObserver } from "../../utils/observers";
+import { Scale, SelectionMode } from "../interfaces";
+import { componentLoaded, setComponentLoaded, setUpLoadableComponent } from "../../utils/loadable";
+/**
+ * @slot - A slot for adding one or more `calcite-card`s.
+ */
+@Component({
+  tag: "calcite-card-group",
+  styleUrl: "card-group.scss",
+  shadow: true,
+})
+export class CardGroup implements InteractiveComponent {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteCardGroupElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /** When `true`, interaction is prevented and the component is displayed with lower opacity. */
+  @Prop({ reflect: true }) disabled = false;
+
+  /** Accessible name for the component. */
+  @Prop() label!: string;
+
+  /** Specifies the size of the component. Child `calcite-card`s inherit the component's value. */
+  @Prop({ reflect: true }) scale: Scale = "m";
+
+  /** Specifies the selection mode of the component. */
+  @Prop({ reflect: true }) selectionMode: Extract<
+    "multiple" | "single" | "single-persist" | "none",
+    SelectionMode
+  > = "none";
+
+  @Watch("selectionMode")
+  onSelectionModeChange(): void {
+    this.updateItems();
+  }
+
+  /**
+   * Specifies the component's selected items.
+   *
+   * @readonly
+   */
+  @Prop({ mutable: true }) selectedItems: HTMLCalciteCardElement[] = [];
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  //--------------------------------------------------------------------------
+
+  mutationObserver = createObserver("mutation", () => this.updateItems());
+
+  private items: HTMLCalciteCardElement[] = [];
+
+  private slotRefEl: HTMLSlotElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  /** Emits when the component's selection changes and the `selectionMode` is not `none`. */
+  @Event({ cancelable: false }) calciteCardGroupSelect: EventEmitter<void>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  connectedCallback(): void {
+    connectInteractive(this);
+    this.mutationObserver?.observe(this.el, { childList: true, subtree: true });
```

Comment:
Agreed w/ @driskull's comment about not needing an observer if items are only to be updated when direct cards are added/removed. 

Putting that aside, if we were to use mutation observer for the above use case, `subtree` isn't needed because it will observe each card's subtree, which I don't think is relevant to item updates.